app/models/products/user_note_mode.rb

```diff
@@ -0,0 +1,74 @@
+module Products
+
+  class UserNoteMode
+
+    VALID_MODES = %w(hidden optional required).freeze
+
+    attr_reader :raw_value
+
+    # An array of UserNoteModes representing the possible modes
+    # Use this in validations and form collections.
+    def self.all
+      VALID_MODES.map { |value| new(value) }
+    end
+
+    def self.from_string(mode_string)
```

Comment:
And which doesn't let you call it from an arbitrary string, so maybe it's not useful at all...

using `UserNoteMode["required"]` is also fine.